akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowInterleaveSpec.scala

```diff
@@ -44,6 +44,15 @@ class FlowInterleaveSpec extends BaseTwoStreamsSetup {
       probe.expectComplete()
     }
 
+    "complete upstream if any of its upstream completes with eagerClose = true" in assertAllStagesStopped {
+      val probe = TestSubscriber.manualProbe[Int]()
+
+      Source(0 to 2).interleave(Source(3 to 5), 2, eagerClose = true).runWith(Sink.fromSubscriber(probe))
+      probe.expectSubscription().request(10)
+      probe.expectNext(0, 1, 3, 4, 2)
+      probe.expectComplete()
+    }
```

Comment:
Would be good to also have test coverage where:
* the other stream closes (probably no issues here, but for regression coverage)
* the source that completes does not do it at a `segmentSize` boundary